extensions/quick-surf/src/input.ts

```diff
@@ -0,0 +1,99 @@
+import { getSelectedText, Clipboard } from "@raycast/api";
+import { isEmailGroup, isEmpty, isMailTo, isUrl, mailtoBuilder, urlIPBuilder } from "./utils";
+import { isIP } from "net";
+
+export enum ItemType {
+  TEXT = "Text",
+  URL = "URL",
+  EMAIL = "Email",
+  NULL = "",
+}
+
+export enum ItemSource {
+  SELECTED = "Selected",
+  CLIPBOARD = "Clipboard",
+  ENTER = "Enter",
+  NULL = "",
+}
+
+export class ItemInput {
+  content: string;
+  source: ItemSource;
+  type: ItemType;
+
+  constructor(content = "", source = ItemSource.NULL, type = ItemType.NULL) {
+    this.content = content;
+    this.source = source;
+    this.type = type;
+  }
+
+  setContent(content = "") {
+    this.content = content;
+    return this;
+  }
+  setSource(source: ItemSource = ItemSource.NULL) {
+    this.source = source;
+    return this;
+  }
+  setType() {
+    const trimText = this.content.trim();
+    if (isEmpty(trimText)) {
+      this.content = "";
+      this.source = ItemSource.NULL;
+      this.type = ItemType.NULL;
+    } else {
+      if (isMailTo(trimText)) {
+        this.content = mailtoBuilder(trimText.slice(7));
+        this.type = ItemType.EMAIL;
+      } else if (isEmailGroup(trimText)) {
+        this.content = mailtoBuilder(trimText);
+        this.type = ItemType.EMAIL;
+      } else if (isUrl(trimText)) {
+        let url;
+        if (isIP(trimText)) {
+          if (trimText == "127.0.0.1") {
+            url = "http://www." + trimText;
+          } else {
+            url = urlIPBuilder("http://", trimText);
```

Comment:
Is it on purpose that this is not using https?

Reply:
Hi @pernielsentikaer 👋. Thank you for the review
The code here is for IP addresses, which are usually prefixed with http, so http is used here instead of https.
When a URL is constructed using a domain name, https is used.